src/bopforge/continuous_pipeline/functions.py

```diff
@@ -27,45 +64,44 @@ def get_signal_model(settings: dict, df: DataFrame) -> MRBeRT:
         Ensemble version of the mrbrt model.
 
     """
+    settings = all_settings["fit_signal_model"]
+    data_settings = _get_data_settings(all_settings)
     data = MRData()
     data.load_df(
         df,
-        col_obs="ln_rr",
-        col_obs_se="ln_rr_se",
+        col_obs=data_settings["obs"],
+        col_obs_se=data_settings["obs_se"],
         col_covs=[
             "ref_risk_lower",
             "ref_risk_upper",
             "alt_risk_lower",
             "alt_risk_upper",
         ],
-        col_study_id="study_id",
-        col_data_id="seq",
+        col_study_id=data_settings["study_id"],
+        col_data_id=data_settings["data_id"],
     )
-    settings["cov_model"] = {
-        **dict(
-            use_re=False,
-            use_spline=True,
-            prior_spline_funval_uniform=[-0.999999, 19],
-            prior_spline_num_constraint_points=100,
-        ),
-        **settings["cov_model"],
-    }
-    cov_model = LogCovModel(
+    cov_model_settings = _get_cov_model_settings(all_settings)
+    type_name = cov_model_settings.pop("type")
+    if type_name == "log":
+        type_class = LogCovModel
+    elif type_name == "linear":
+        type_class = LinearCovModel
+    else:
+        raise TypeError(
+            f"Unrecognized cov_model type='{type_name}', can only choose from "
+            "'log' or 'linear'"
+        )
```

Comment:
Do we also need corresponding changes in the mrtool.core.cov_model?

Reply:
No, the program will choose `LogCovModel` and `LinearCovModel` based on the `type` specified, we don't need to touch `mrtool` for this :)